### PR TITLE
Fix the inconsistency bug for mode in subpopulation

### PR DESCRIPTION
**Pull request by syhwawa**
In the `scoringParameter` section in the matsim config, sometimes the `mode`  in `modeParams` for different subpopulations is not consistent with the subpopulations. e.g for `hgv` only has `car` as the  `modeParams`. 

This PR fixed this inconsistency bug when requiring the scoring values for various modes in different subpopulations by returning the None if the mode is not found in this subpopulation. 

```
=======================================parameters for scoring=======================================
===========================================subpopulation============================================
subpopulation: default,hhs_carAvailyes,hhs_carAvailnever,hgv,
=======scoring parameters for subpopulation=======
marginalUtilityOfMoney:10,10,10,10,
performing:100,100,100,100,
---------------------mode:bus---------------------
mode_specific_constant:-1,-1,-1,NA
marginal_utility_of_distance:0.0,0.0,0.0,NA
marginal_utility_of_traveling:-2,-2,-2,NA
monetary_distance_rate:-0.00036832,-0.00036832,-0.00036832,NA
```

Add a date at the head of log. 

Add a more realistic NZ matsim config for testing in the future. 

The updated mc has been tested via bitsim orchestration in the NZ project.

### CODE REVIEW

**Review by fredshone — APPROVED** [review]
happy for you to merge this. From a code style point of view it is slightly frowned upon to call something by it's type, ie `dict1`. The reader knows it's a dictionary already so better to call it something more informative, like `scoring_record`.

File: mc/summarise.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import os
+from datetime import date
 
 
 def diretory_log_summary(config):
```
Comment (rorysedgwick) [inline mc/summarise.py:5]: I think this should be spelled `directory_`

**Review by rorysedgwick — APPROVED** [review]

**Comment by rorysedgwick** [comment]
👍  that it's been tested in the wild already, and confirmed working.
Worth flagging that it might be easier/quicker to test changes in Matesto if you want to quickly run a few jobs in sequence

File: mc/summarise.py (hunk 1 above)
Comment (syhwawa) [reply mc/summarise.py]: typo fixed.

**Comment by syhwawa** [comment]
> happy for you to merge this. From a code style point of view it is slightly frowned upon to call something by it's type, ie `dict1`. The reader knows it's a dictionary already so better to call it something more informative, like `scoring_record`.

Good point. It has been changed.

**Pull request merged**